src/bootstrap/builder.rs

```diff
@@ -850,6 +855,21 @@ impl<'a> Builder<'a> {
         Self::new_internal(build, kind, paths.to_owned())
     }
 
+    /// Creates a new standalone builder for use outside of the normal process
+    pub fn new_standalone(
+        build: &mut Build,
+        kind: Kind,
+        paths: Vec<PathBuf>,
+        stage: Option<u32>,
+    ) -> Builder<'_> {
+        // FIXME: don't mutate `build`
+        if let Some(stage) = stage {
+            build.config.stage = stage;
+        }
+
+        Self::new_internal(build, kind, paths.to_owned())
```

Comment:
I think you can avoid needing this new function by modifying `build.config.cmd` and using `Builder::new` instead. You could change `sug!` to generate the command for you so it's not too much of a pain:
```rust
(test, $stage:literal, $paths:expr) => Suggestion { stage: $stage, cmd: Subcommand::Test { paths: $paths } },
```

Reply:
Thanks for the idea, I think I also need to modify `build.config.stage`. Sorry I'm on holiday right now so the progress is slow.

Reply:
No need to apologize! You're responding quite fast actually :smile:

Reply:
Hmm, using `serde_json` means that we can't directly create `Subcommand`s. Honestly, the solution is good enough as is for now.

Reply:
I'm not sure I understand the problem, sorry - how is serde_json related here?

Reply:
@Ezrashaw I don't think I ever got an answer here - how is serde related?

Reply:
> Hmm, using serde_json means that we can't directly create Subcommands. Honestly, the solution is good enough as is for now.

Sorry about that. I have no idea what I was thinking lol. I think that this isn't that relevant anymore since the code was moved to a separate crate.